Feat : DIG-161 운동일지 웹푸시 알람보내기

YEJINGO opened the pull request:
- build.gradle : FCM 푸시를 위한 의존성 추가
- 로그인 시, fcm 토큰을 발급받고 fcmRepository에 fcm토큰이 없을 경우에만 저장
- 지정된 시간에 미리 작성된 운동일지의 운동에 대한 알림을 보냄 ex) 오늘은 푸쉬업, 어시스트 풀업 머신 하는 날입니다.
    - 내일(24일 금요일) 오후 3시에 테스트해보기 위해 3시로 설정해놓음 
    - 추후 오전 7시30분으로 수정할 예정
- 로그아웃 시, 유저정보를 받아와 저장된 fcm 토큰을 삭제함

dunowljj reviewed (approved):
고생하셨어요~

dunowljj commented on src/main/java/com/ogjg/daitgym/alarm/service/FcmAlarmService.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    private User getUser(OAuth2JwtUserDetails oAuth2JwtUserDetails) {
+        return userRepository.findByEmail(oAuth2JwtUserDetails.getEmail())
+                .orElseThrow(NotFoundUser::new);
```
조회 로직이 helper 클래스에 있긴한데..! 필요하면 재활용하셔도 될거같습니다!

YEJINGO replied:
감사합니다!! 수정했습니다

YEJINGO committed:
Feat : DIG-161 운동일지 웹푸시 알람보내기

- build.gradle : FCM 푸시를 위한 의존성 추가
- 로그인 시, fcm 토큰을 발급받고 fcmRepository에 fcm토큰이 없을 경우에만 저장
- 지정된 시간에 미리 작성된 운동일지의 운동에 대한 알림을 보냄 ex) 오늘은 푸쉬업, 어시스트 풀업 머신 하는 날입니다.
    - 내일(24일 금요일) 오후 3시에 테스트해보기 위해 3시로 설정해놓음
    - 추후 오전 7시30분으로 수정할 예정
- 로그아웃 시, 유저정보를 받아와 저장된 fcm 토큰을 삭제함
- DaItGymApplication에 스케쥴링 사용을 위해 @EnableScheduling 추가
- gitignore : daitgym-firebase-sdk.json 추가
